### PR TITLE
feat: Update resources to Karpenter v1 #44

### LINKED ISSUE #44: Karpenter v1 support
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### What is the outcome that you are trying to reach?

Update the NodePool and EC2NodeClass resources for Karpenter v1

### PR DESCRIPTION
**Pull request by chrismld**
### What does this PR do?

Updating the Karpenter resources to v1 following the guidance from this blog: https://aws.amazon.com/blogs/containers/announcing-karpenter-1-0/

### Motivation

- Resolves #44

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I ran `pre-commit run -a` with this PR

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by vara-bonthu — APPROVED** [review]
LGTM! Please confirm the test is complete on the latest code. 

We will create a new release for this module so that existing users are not impacted by this breaking change.

**Comment by chrismld** [comment]
@vara-bonthu I tested with the updated version for the Kafka DoEKS blueprint and found a few minor issues.

Here are the results of the tests, the NodePools and EC2NodeClasses were deployed without problems:

```
➜ kubectl get nodes
NAME                                          STATUS   ROLES    AGE    VERSION
ip-100-64-118-20.eu-west-1.compute.internal   Ready    <none>   150m   v1.31.0-eks-a737599
ip-100-64-212-93.eu-west-1.compute.internal   Ready    <none>   150m   v1.31.0-eks-a737599
ip-100-64-33-48.eu-west-1.compute.internal    Ready    <none>   150m   v1.31.0-eks-a737599

➜ kubectl get nodepools
NAME      NODECLASS   NODES   READY   AGE
default   default     0       True    11m
kafka     kafka       0       True    11m

➜ kubectl get ec2nodeclass
NAME      READY   AGE
default   True    11m
kafka     True    11m

➜ helm ls --all-namespaces
NAME                 	NAMESPACE             	REVISION	UPDATED                              	STATUS  	CHART                        	APP VERSION
default              	karpenter-resources   	1       	2024-10-03 22:14:58.348681 +0200 CEST	deployed	karpenter-resources-0.0.2    	0.0.2
kafka                	karpenter-resources   	1       	2024-10-03 22:14:58.348632 +0200 CEST	deployed	karpenter-resources-0.0.2    	0.0.2
karpenter            	karpenter             	14      	2024-10-03 22:24:02.76809 +0200 CEST 	deployed	karpenter-1.0.5              	1.0.5
kube-prometheus-stack	kube-prometheus-stack 	1       	2024-10-03 21:45:14.860397 +0200 CEST	deployed	kube-prometheus-stack-48.1.1 	v0.66.0
metrics-server       	kube-system           	1       	2024-10-03 21:45:07.717261 +0200 CEST	deployed	metrics-server-3.12.0        	0.7.0
strimzi-operator     	strimzi-kafka-operator	1       	2024-10-03 21:46:30.666869 +0200 CEST	deployed	strimzi-kafka-operator-0.43.0	0.43.0

➜ kubectl get pods -n kafka
NAME                                       READY   STATUS    RESTARTS   AGE
cluster-broker-0                           1/1     Running   0          3m23s
cluster-broker-1                           1/1     Running   0          3m23s
cluster-broker-2                           1/1     Running   0          3m23s
cluster-controller-3                       1/1     Running   0          3m23s
cluster-controller-4                       1/1     Running   0          3m23s
cluster-controller-5                       1/1     Running   0          3m23s
cluster-cruise-control-67646995c4-4zx9z    1/1     Running   0          56s
cluster-entity-operator-6d9d688fdb-8jmb9   2/2     Running   0          77s
cluster-kafka-exporter-5897d5c9b5-rv9kt    1/1     Running   0          34s

➜ kubectl get nodeclaims
NAME          TYPE          CAPACITY    ZONE         NODE                                          READY   AGE
kafka-6qwfp   r6g.4xlarge   on-demand   eu-west-1c   ip-100-64-91-111.eu-west-1.compute.internal   True    3m27s
kafka-7889l   r6g.4xlarge   on-demand   eu-west-1a   ip-100-64-248-86.eu-west-1.compute.internal   True    3m17s
kafka-bkcvb   r6g.4xlarge   on-demand   eu-west-1c   ip-100-64-120-60.eu-west-1.compute.internal   True    2m57s
kafka-jgll7   r6g.4xlarge   on-demand   eu-west-1b   ip-10-1-2-240.eu-west-1.compute.internal      True    3m7s
kafka-nfjz9   r6g.4xlarge   on-demand   eu-west-1a   ip-100-64-179-83.eu-west-1.compute.internal   True    3m27s
kafka-nj7z4   r6g.4xlarge   on-demand   eu-west-1b   ip-10-1-2-147.eu-west-1.compute.internal      True    3m17s
```

File: helm-charts/karpenter-resources/values.yaml [hunk 1]
```diff
@@ -5,9 +5,9 @@ clusterName: test-cluster
 # EC2NodeClass specific values
 ec2NodeClass:
   enabled: true
-  amiFamily: Bottlerocket
+  amiFamily: AL2
```
Comment (bryantbiggs) [inline helm-charts/karpenter-resources/values.yaml:8]: we should not use AL2 as a default anymore - either Bottlerocket or AL2023

Comment (bryantbiggs) [reply helm-charts/karpenter-resources/values.yaml:8]: If we are updating patterns, then we should update to either Bottlerocket or AL2023 - AL2 is reaching the end of support

Comment (chrismld) [reply helm-charts/karpenter-resources/values.yaml:8]: @bryantbiggs makes sense to me ... @vara-bonthu what do you think? Testing other blueprints with AL2023 or Bottlerocket will take time, and as we're using versioning, we won't break any existing DoEKS blueprint but we would need to let other owners know about it (I can help with that).

As soon as we have an agreement, I'll push a change to use Bottlerocket as default (or AL2023 if there's a preference on that).

Comment (chrismld) [reply helm-charts/karpenter-resources/values.yaml]: @vara-bonthu I've pushed a new commit to have Bottlerocket as a default amiSelectorTerms

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Pull request merged**